cmd/zed/agents/zfs_retire.c

```diff
@@ -343,11 +406,29 @@ zfs_retire_recv(fmd_hdl_t *hdl, fmd_event_t *ep, nvlist_t *nvl,
 		char *devtype;
 		char *devname;
 
+		if (nvlist_lookup_string(nvl, FM_EREPORT_PAYLOAD_ZFS_VDEV_TYPE,
+		    &devtype) == 0) {
+			if (strcmp(devtype, VDEV_TYPE_SPARE) == 0)
+				spare = 1;
+			else if (strcmp(devtype, VDEV_TYPE_L2CACHE) == 0)
+				l2arc = 1;
+		}
+
+		if (nvlist_lookup_uint64(nvl,
+		    FM_EREPORT_PAYLOAD_ZFS_VDEV_GUID, &vdev_guid) != 0) {
+			return;
+		}
```

Comment:
We typically omit braces for a single line in ZFS, as seen elsewhere in this function.

Reply:
Thank you for pointing that out. I wasn't very much sure about multiline conditions, updated it.